### PR TITLE
#35568 Zero warnings

### PR DESCRIPTION
**Pull request by svick**

### CODE REVIEW

File: eng/src/BuildMetalamaSamples.csproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
         <GenerateDocumentationFile>true</GenerateDocumentationFile>
         <LangVersion>latest</LangVersion>
         <Nullable>enable</Nullable>
-        <NoWarn>SA0001;CS8002;SA1114;SA1115</NoWarn>
+        <NoWarn>SA0001;CS8002;SA1114;SA1115;NU1903;NU1904</NoWarn>
```
Comment (prochan2) [inline eng/src/BuildMetalamaSamples.csproj]: Comment on what these warnings mean, why they are suppressed, and when (if ever) this suppression should be removed.

File: examples/builder/builder-2/GenerateBuilderAttribute.cs [hunk 2]
```diff
@@ -119,7 +118,7 @@ public override void BuildAspect(IAspectBuilder<INamedType> builder)
                 {
                     var isRequired = p.Attributes.OfAttributeType(typeof(RequiredAttribute))
                         .Any();
-                    var isInherited = p.DeclaringType != sourceType;
+                    var isInherited = !p.DeclaringType.Equals(sourceType);
```
Comment (prochan2) [inline examples/builder/builder-2/GenerateBuilderAttribute.cs]: Why?

File: examples/exception-handling/retry/retry-5/ResiliencePipelineFactory.cs [hunk 3]
```diff
@@ -19,7 +19,6 @@ public ResiliencePipeline GetPipeline( StrategyKind strategyKind )
                             BackoffType = DelayBackoffType.Exponential,
                             MaxRetryAttempts = 10
                         } );
```
Comment (prochan2) [inline examples/exception-handling/retry/retry-5/ResiliencePipelineFactory.cs]: Inconsistent formatting. The spaces around brackets are kept in some places and removed in others. Check that the code style is set up correctly. See https://github.com/postsharp/postSharp.Engineering.codestyle.

**Review by prochan2 — CHANGES_REQUESTED** [review]

File: examples/builder/builder-3/PropertyMappingFactory.cs [hunk 4]
```diff
@@ -8,7 +8,7 @@ namespace Metalama.Samples.Builder3;
 [CompileTime]
 internal class PropertyMappingFactory
 {
-    private static readonly HashSet<string> _immutableCollectionTypes =
+    private static readonly HashSet<string> ImmutableCollectionTypes =
```
Comment (gfraiteur) [inline examples/builder/builder-3/PropertyMappingFactory.cs]: The initial casing was the right one.

File: examples/caching/caching-4/CachingOptionsExtensions.cs [hunk 5]
```diff
@@ -11,25 +11,25 @@ public static class CachingOptionsExtensions
 {
     private static readonly ConditionalWeakTable<CachingOptions,
             ImmutableDictionary<string, CacheBuilderRegistration>>
-        _cache = new();
+        Cache = new();
```
Comment (gfraiteur) [inline examples/caching/caching-4/CachingOptionsExtensions.cs]: The initial casing was the right one.

File: examples/caching/caching-4/CachingOptionsExtensions.cs [hunk 6]
```diff
@@ -11,25 +11,25 @@ public static class CachingOptionsExtensions
...
 
 
-    private static readonly DiagnosticDefinition<IType> _error = new("CACHE01", Severity.Error,
+    private static readonly DiagnosticDefinition<IType> Error = new("CACHE01", Severity.Error,
```
Comment (gfraiteur) [inline examples/caching/caching-4/CachingOptionsExtensions.cs]: The initial casing was the right one.

File: examples/clone/clone-2/CloneableAttribute.cs [hunk 7]
```diff
@@ -66,16 +66,16 @@ private bool VerifyFieldsAndProperties(IAspectBuilder<INamedType> builder)
             if (fieldOrProperty.Writeability != Writeability.All)
             {
                 builder.Diagnostics.Report(
-                    _fieldOrPropertyCannotBeReadOnly.WithArguments((
+                    FieldOrPropertyCannotBeReadOnly.WithArguments((
```
Comment (gfraiteur) [inline examples/clone/clone-2/CloneableAttribute.cs]: It seems all these case changes must be reverted, which makes most of the PR.

File: examples/exception-handling/retry/retry-5/ResiliencePipelineFactory.cs (hunk 3 above)
Comment (svick) [reply examples/exception-handling/retry/retry-5/ResiliencePipelineFactory.cs]: I ran `b codestyle pull`, fixed path to CommonStyle, and ran `b codestyle format` (in a separate commit).

File: examples/builder/builder-2/GenerateBuilderAttribute.cs (hunk 2 above)
Comment (svick) [reply examples/builder/builder-2/GenerateBuilderAttribute.cs]: It's generally better to compare `IType`s using `Equals()` than using reference equality.

**Review by prochan2 — APPROVED** [review]

**Pull request merged**